nui/src/main/java/org/terasology/nui/widgets/types/TypeWidgetFactoryRegistry.java

```diff
@@ -0,0 +1,12 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.nui.widgets.types;
+
+import java.util.List;
+
+public interface TypeWidgetFactoryRegistry {
```

Comment:
what's the difference between a Registry and a Library?

Reply:
A `TypeWidgetFactoryRegistry` is, simply put, a list of `TypeWidgetFactory`s. A `TypeWidgetLibrary`, on the other hand, generates widgets for arbitrary types. Internally, it may use a list of `TypeWidgetFactory`s (or a `TypeWidgetFactoryRegistry`) to perform the generation.